sunpy/net/helioviewer.py

```diff
@@ -450,3 +499,17 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, source_id, key):
+        if source_id is None:
+            try:
+                source_id_list = list(partial_match(key,self.data_sources))
+                if len(source_id_list) == 1:
```

Comment:
I think if a partial match is found that is >1, we should add that to the `raise` so the user knows what was found.

Reply:
Shall we raise a KeyError or a ValueError saying that `Got more values than expected` and then just showing the list of all the values returned by the function?

Reply:
I would say its still a KeyError but we can change it if we need to later.